### PR TITLE
DEV-640 Refactored locations api description

### PR DESCRIPTION
**Pull request by christiangoltz**
This pull request contains the next_available_slot, current, maximum and availability changes introduced with version 4 of the apps.

I also updated the underlying infrastructure a bit and changed table borders to be visible for a better overview in long tables.

### CODE REVIEW

File: api/locations.md [hunk 1]
```diff
@@ -22,42 +22,41 @@ $ curl {{ site.parku.api }}/locations/00cd7cfd-e42d-11e2-8bf1-8a83f3373875 \
...
+    "id": "00cd7cfd-e42d-11e2-8bf1-8a83f3373875",
+    "type": "off-street",
+    "code": "ZHRTHW",
     "description": "Doppelparkplatz. Jedoch nur für ein Auto zu vermieten. Die andere Hälfte wird benötigt. Aufteilung wie im Bild.",
+    "latitude": 52.505681,
+    "longitude": 13.303608,
     "image_urls": [
-        "http://parku.ch/parkingspace/00cd7cfd-e42d-11e2-8bf1-8a83f3373875/image",
-        "http://parku.ch/parkingspace/00cd7cfd-e42d-11e2-8bf1-8a83f3373875/image"
+        "https://parku.de/images/parkingspace/020a81f5-c8b9-11e4-bfd2-d43d7eece658/640x286.jpg"
```
Comment (martinsey) [inline api/locations.md]: Did this change in DEV-640, or is it just a general update?

File: api/locations.md [hunk 2]
```diff
@@ -86,58 +92,71 @@ Parameter | Description
...
+200  | location found, body contains description
+404  | location was not found, check the location id and that your user has the correct access rights to see the location
+422  | request validation failed because the request data was malformed
+
+#### Body
+Field             | Element                | Type         | (Element and) Description
+----------------- | ---------------------- | ------------ | -----
+`address_slug`    |                        | `string`     | street name and number of the location in url address format
+`availability`    |                        | `dictionary` | describes the availability of the location in relation to the request
+                  | `current`              | `number`     | the currently available number of parking spaces at the location for the requested time period defined by `start` and `end`
```
Comment (martinsey) [inline api/locations.md]: I would remove "currently available" and change it to something like "available for timeframe"

File: api/locations.md [hunk 3]
```diff
@@ -86,58 +92,71 @@ Parameter | Description
...
+                  | `maximum`              | `number`     | the overall number of parking spaces at this location
+                  | `next_start`           | `time`       | the start of the next from now slot in which a parking space in this location is free and bookable, can be in the past
+                  | `next_end`             | `time`       | the end of the next slot from now in which a parking space in this location is free and bookable, can be in the past
+`city`            |                        | `string`     | city of the location
+`code`            |                        | `string`     | short identification code of the location, not necessarily unique
+`country`         |                        | `string`     | country of the location in 2 character encoding
+`currency`        |                        | `string`     | the currency that the price for the location is in, `EUR` for Euro or `CHF` for swiss franc.
+`description`     |                        | `string`     | the description of the location. The description is
```
Comment (martinsey) [inline api/locations.md]: There is something missing in that sentence, right?

File: api/locations.md [hunk 4]
```diff
@@ -86,58 +92,71 @@ Parameter | Description
...
+`settings`        |                        | `dictionary` | provides booking requirements for the location. The settings of a location are used in favor of the global settings that can be found in the settings endpoint.
+                  | `bookable_range_start` | `number`     | start of the range wherein a booking is allowed in minutes from now.
+                  | `bookable_range_end`   | `number`     | end of the range wherein a booking is allowed in minutes from now.
+                  | `booking_duration_min` | `number`     | minimum duration of a booking in minutes.
+                  | `booking_duration_max` | `number`     | maximum duration of a booking in minutes.
+`street`          |                        | `string`     | the street name of the location without number
+`street_number`   |                        | `string`     | the street number of the location
+`support`         |                        | `dictionary` | contains field `phone_number` that returns the support center phone number for this location
```
Comment (martinsey) [inline api/locations.md]: Can this field contain multiple entries?

File: api/locations.md [hunk 5]
```diff
@@ -282,15 +312,23 @@ Status: 200 OK
...
-`sw` | Comma separated latitude and longitude of the southwest position. _Required. Will become optional._
-`ne` | Comma separated latitude and longitude of the northeast position. _Required. Will become optional._
-`date_start` | Filter parking spaces for the start date. _Optional._
-`date_end` | Filter parking spaces for the end date. _Optional._
+Parameter      | Value      | Description
+-------------- | ---------- | ----
+`sw`           | `decimal`  |Comma separated latitude and longitude of the southwest position. _Required.
+`ne`           | `decimal`  |Comma separated latitude and longitude of the northeast position. _Required.
+`date_start`   | `datetime` | Filter parking spaces for the start date. _Optional._
+`date_end`     | `datetime` | Filter parking spaces for the end date. _Optional._
```
Comment (martinsey) [inline api/locations.md]: If `date_start` is provided, `date_end` is necessary?

File: api/locations.md [hunk 6]
```diff
@@ -282,15 +312,23 @@ Status: 200 OK
...
-Parameter | Description
---- | ---
-`sw` | Comma separated latitude and longitude of the southwest position. _Required. Will become optional._
-`ne` | Comma separated latitude and longitude of the northeast position. _Required. Will become optional._
-`date_start` | Filter parking spaces for the start date. _Optional._
-`date_end` | Filter parking spaces for the end date. _Optional._
+Parameter      | Value      | Description
+-------------- | ---------- | ----
+`sw`           | `decimal`  |Comma separated latitude and longitude of the southwest position. _Required.
+`ne`           | `decimal`  |Comma separated latitude and longitude of the northeast position. _Required.
```
Comment (martinsey) [inline api/locations.md]: Why is `ne` italic, while `sw` is not?

File: api/locations.md (hunk 1 above)
Comment (christiangoltz) [inline api/locations.md]: didnt change in dev-640, rather earlier I think, but this is a recent example.

File: api/locations.md (hunk 2 above)
Comment (christiangoltz) [inline api/locations.md]: changed to `the number of parking spaces available at the location for the requested time period defined by `start` and `end``

File: api/locations.md (hunk 3 above)
Comment (christiangoltz) [inline api/locations.md]: yes the information about translation was missing, added that one

File: api/locations.md (hunk 4 above)
Comment (christiangoltz) [inline api/locations.md]: checked that before, it cannot

File: api/locations.md (hunk 6 above)
Comment (christiangoltz) [inline api/locations.md]: Issue with the editor, forgot the second _ after Required, fixed

File: api/locations.md (hunk 5 above)
Comment (christiangoltz) [inline api/locations.md]: From a form perspective both are optional, if you provide only date_start without date_end it should work as if date_start was not provided. added a comment for that

**Pull request merged**